internal/util/file.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2022 The Flux authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+)
+
+func writeBytesToFile(bytes []byte, out string, temp bool) (*os.File, error) {
```

Comment:
Now that you've two wrapper functions below, how about creating the file in the wrapper function and don't need to pass a `temp` variable to this function, pass it `os.File` can you can just write directly to. Since the wrapper function created the file, it already has the `os.File`, this function can just return an error.